pkg/volume/gce_pd/gce_util.go

```diff
@@ -184,13 +168,51 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 			int64(requestGB),
 			*c.options.CloudTags); err != nil {
 			glog.V(2).Infof("Error creating single-zone GCE PD volume: %v", err)
-			return "", 0, nil, "", err
+			return err
 		}
 
 		glog.V(2).Infof("Successfully created single-zone GCE PD volume %s", name)
+		return nil
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
```

Comment:
Is this check really necessary, if you've already gated the possible values of `replicationType` above?

Reply:
True. I added the check to be clear exactly which branch the program takes if the gate is set vs unset, but it does hinder readability. Will remove.